lib/auto_attach.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+
+"""
+Perform auto-attach operation
+
+On Ubuntu Pro machines, we try to perform an auto-attach operation
+on first boot. This happens through a systemd unit that triggers
+that executes this script on every boot. However, if we detect
+that cloud-init has user data related to ua, we cannot run
+auto-attach here, since cloud-init will drive this operation on
+their side.
+"""
+import logging
+import sys
+
+from uaclient.cli import action_auto_attach, setup_logging
+from uaclient.config import UAConfig
+from uaclient.exceptions import AlreadyAttachedOnPROError
+
+try:
+    import cloudinit.stages as ci_stages  # type: ignore
+except ImportError:
+    pass
+
+
+def get_cloudinit_init_stage():
+    if "cloudinit.stages" in sys.modules:
+        return ci_stages.Init()
+
+    return None
```

Comment:
instead of the check in sys.modules, would it be better or worse to do a runtime import?
```suggestion
def get_cloudinit_init_stage():
    try:
        import cloudinit.stages as ci_stages  # type: ignore
        return ci_stages.Init()
    except ImportError:
        return None
```

Reply:
Yeah, that was my original idea. The problem is that if we import cloudinit like this we end up messing our logging, because this cloudinit module seems to override our setup. But maybe I missed another thing we could here to not mess up the logs

Reply:
Oh the import is modifying root logger settings? Then the current implementation makes sense